Failure to parse function pointer type typedef properly

```
    FAILED: _rizin.cpython-38-x86_64-linux-gnu.so.p/meson-generated_.._rizin_wrap.cxx.o
    c++ -I_rizin.cpython-38-x86_64-linux-gnu.so.p -I. -I.. -I/opt/_internal/cpython-3.8.20/include/python3.8 -I/usr/local/include/librz -I/usr/local/include/librz/sdb -fdiagnostics-color=always -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O3 -fPIC -MD -MQ _rizin.cpython-38-x86_64-linux-gnu.so.p/meson-generated_.._rizin_wrap.cxx.o -MF _rizin.cpython-38-x86_64-linux-gnu.so.p/meson-generated_.._rizin_wrap.cxx.o.d -o _rizin.cpython-38-x86_64-linux-gnu.so.p/meson-generated_.._rizin_wrap.cxx.o -c rizin_wrap.cxx
    rizin_wrap.cxx: In function 'PyObject* _wrap_RzBuffer_fwd_scan(PyObject*, PyObject*)':
    rizin_wrap.cxx:77903:31: error: invalid operands of types 'long long unsigned int' and 'std::nullptr_t' to binary 'operator!='
    77903 |   SWIG_contract_assert((result!=NULL), "Contract violation: require: (result!=NULL)");
          |                               ^
    rizin_wrap.cxx:3153:52: note: in definition of macro 'SWIG_contract_assert'
     3153 | #define SWIG_contract_assert(expr, msg) do { if (!(expr)) { SWIG_Error(SWIG_RuntimeError, msg); SWIG_fail; } } while (0)
          |                                                    ^~~~
```

If we check the function in question, it has this code:
```c
typedef ut64 (*RzBufferFwdScan)(RZ_BORROW RZ_NONNULL const ut8 *buf, ut64 len, RZ_NULLABLE void *user);
RZ_API ut64 rz_buf_fwd_scan(RZ_NONNULL RzBuffer *b, ut64 start, ut64 amount, RZ_NONNULL RzBufferFwdScan fwd_scan, RZ_NULLABLE void *user);

/**
 * \brief Scans buffer linearly in chunks calling \p fwd_scan for each chunk.
 *
 * \param b RzBuffer to read
 * \param start Start address
 * \param amount Amount of bytes to read
 * \param fwd_scan Function to call for each chunk
 * \param user User data to pass to fwd_scan
 * \return Number of bytes read
 */
RZ_API ut64 rz_buf_fwd_scan(RZ_NONNULL RzBuffer *b, ut64 start, ut64 amount, RZ_NONNULL RzBufferFwdScan fwd_scan, RZ_NULLABLE void *user) {
	rz_return_val_if_fail(b && fwd_scan, 0);
	if (!amount) {
		return 0;
	}
       ...
}
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -16,9 +16,13 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-22.04, windows-2019, macos-12]
+        os: [ubuntu-22.04, windows-2019, macos-14]
 
     steps:
+    - name: Install SWIG with Homebrew
+      if: matrix.os == 'macos-14' && matrix.enabled
+      run: brew install swig
+
     - name: Checkout rz-bindgen
       uses: actions/checkout@v4
 

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -30,7 +30,7 @@ jobs:
         ref: ${{ inputs.rizin_ref }}
 
     - name: Build rz-bindgen
-      uses: pypa/cibuildwheel@v2.17.0
+      uses: pypa/cibuildwheel@v2.22.0
 
     - name: Upload sdist
       uses: actions/upload-artifact@v4

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -66,7 +66,13 @@ jobs:
     - name: Install dependencies
       run: |
         sudo apt update
-        sudo apt-get --assume-yes install cmake swig pkg-config clang libclang-dev llvm wget unzip python3-wheel python3-setuptools build-essential python3-pip && sudo pip3 install meson ninja
+        sudo apt-get --assume-yes install cmake swig pkg-config clang libclang-dev llvm wget unzip python3-wheel python3-setuptools build-essential python3-pip
+        pygt311=$(python3 -c 'import sys; print(0) if sys.version_info.minor > 11 else print(1)' 2>&1)
+        if [[ $pygt311 == '0' ]]; then
+          sudo pip3 --break-system-packages install meson ninja
+        else
+          sudo pip3 install meson ninja
+        fi
 
     - name: Install rizin
       run: |

--- a/cibw_before_all.sh
+++ b/cibw_before_all.sh
@@ -6,21 +6,23 @@ set -ex
 # Install deps
 pip3 install meson ninja meson-python build
 
-if command -v apt; then
+if command -v brew; then
+    brew install --require-sha swig
+elif command -v apt; then
     set +e
     grep VERSION_CODENAME=stretch /etc/os-release >/dev/null 2>/dev/null
     if [[ $? == 0 ]] ; then
-	# Required because manylinux_2_24 uses Debian Stretch and the repos are not available anymore
+        # Required because manylinux_2_24 uses Debian Stretch and the repos are not available anymore
         echo "deb http://archive.debian.org/debian stretch main" > /etc/apt/sources.list
         echo "deb http://archive.debian.org/debian-security stretch/updates main" >> /etc/apt/sources.list
         echo "Acquire::Check-Valid-Until no;" > /etc/apt/apt.conf.d/99no-check-valid-until
     fi
     set -e
-    apt update && apt install --assume-yes libclang-7-dev clang-7 llvm-7
+    apt update && apt install --assume-yes libclang-7-dev clang-7 llvm-7 sudo
 elif command -v apk; then
-    apk update && apk add clang-dev
+    apk update && apk add clang-dev sudo
 elif command -v yum; then
-    yum -y install llvm-toolset-7.0 llvm-toolset-7.0-llvm-devel
+    yum -y install llvm-toolset-7.0 llvm-toolset-7.0-llvm-devel sudo
     # (CentOS 7) Uncomment following lines to see required paths in build log:
     #   yum -y install centos-release-scl
     #   source scl_source enable llvm-toolset-7.0 || true
@@ -30,11 +32,12 @@ pushd rizin
 
 if [[ "$OSTYPE" =~ msys* ]]; then
     meson setup --buildtype=release --prefix='c:/rizin' --vsenv build
+    meson install -C build
 else
     meson setup --buildtype=release --libdir=lib build
+    sudo meson install -C build
 fi
 
-meson install -C build
 popd
 
 if command -v apt; then

--- a/cibw_before_all.sh
+++ b/cibw_before_all.sh
@@ -4,7 +4,12 @@
 set -ex
 
 # Install deps
-pip3 install meson ninja meson-python build
+pygt311=$(python3 -c 'import sys; print(0) if sys.version_info.minor > 11 else print(1)' 2>&1)
+if [[ $pygt311 == '0' ]]; then
+    pip3 install --break-system-packages meson ninja meson-python build
+else
+    pip3 install meson ninja meson-python build
+fi
 
 if command -v brew; then
     brew install --require-sha swig

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -22,10 +22,13 @@ archs = ["x86_64"]
 environment = { CMAKE_PREFIX_PATH="C:\\rizin" }
 archs = ["AMD64"]
 
+[tool.cibuildwheel.macos]
+environment = { PATH="/opt/homebrew/bin:$PATH", MACOSX_DEPLOYMENT_TARGET="14.0" }
+
 [project]
 name = "rz-bindings"
 dynamic = [
   'version',
 ]
-requires-python = ">=3.8"
+requires-python = ">=3.9"
 readme = "README.md"

--- a/src/bindings.py
+++ b/src/bindings.py
@@ -258,7 +258,6 @@ def bind_cons(cons_h: Header) -> None:
         typedef="RzCons",
         ignore_fields={"term_raw", "term_buf"},  # struct termios
         rename_fields={
-            "lastline": "_lastline",
             "echo": "_echo",
             "highlight": "_highlight",
             "newline": "_newline",

--- a/src/generator_swig.py
+++ b/src/generator_swig.py
@@ -285,7 +285,7 @@ def write_func(writer: Writer, func: Func, name: str, kind: FuncKind) -> None:
 
     # Nullability checking contract
     if args_nonnull:
-        writer.line(f"%contract {name}({args_outer_str}) {{", "require:")
+        writer.line(f"%contract {name}({args_inner_str}) {{", "require:")
         with writer.indent():
             for contract_arg in args_nonnull:
                 writer.line(f"{contract_arg} != NULL;")

